config/v1/types_node.go

```diff
@@ -0,0 +1,156 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Node holds cluster-wide information about node specific features.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+type Node struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec NodeSpec `json:"spec"`
+
+	// status holds observed values.
+	// +optional
+	Status NodeStatus `json:"status"`
+}
+
+type NodeSpec struct {
+	// CgroupMode determines the cgroups version on the node
+	// +optional
+	CgroupMode CgroupMode `json:"cgroupMode,omitempty"`
+
+	// WorkerLatencyProfile determins the how fast the kubelet is updating
+	// the status and corresponding reaction of the cluster
+	// +optional
+	WorkerLatencyProfile WorkerLatencyProfileType `json:"workerLatencyProfile,omitempty"`
+}
+
+type NodeStatus struct {
+	// WorkerLatencyProfileStatus provides the current status of WorkerLatencyProfile
+	// +optional
+	WorkerLatencyProfileStatus WorkerLatencyProfileStatus `json:"workerLatencyProfileStatus,omitempty"`
+}
+
+type CgroupMode string
+
+const (
+	CgroupMode_Empty   = "" // Empty string will always use the Default
+	CgroupMode_v1      = "v1"
+	CgroupMode_v2      = "v2"
+	CgroupMode_Default = CgroupMode_v1
+)
+
+type WorkerLatencyProfileType string
+
+const (
+	// Medium Kubelet Update Frequency (heart-beat) and Average Reaction Time to unresponsive Node
+	MediumUpdateAverageReaction WorkerLatencyProfileType = "MediumUpdateAverageReaction"
+
+	// Low Kubelet Update Frequency (heart-beat) and Slow Reaction Time to unresponsive Node
+	LowUpdateSlowReaction WorkerLatencyProfileType = "LowUpdateSlowReaction"
+
+	// Default values of relavent Kubelet, Kube Controller Manager and Kube API Server
+	DefaultUpdateDefaultReaction WorkerLatencyProfileType = "Default"
+)
+
+// WorkerLatencyProfileStatus provides status information about the WorkerLatencyProfile rollout
+type WorkerLatencyProfileStatus struct {
+	// conditions describes the state of the WorkerLatencyProfile and related components
+	// (Kubelet or Controller Manager or Kube API Server)
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +optional
+	Conditions []WorkerLatencyStatusCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+	// relatedObjects is a list of objects that are "interesting" or related to this WorkerLatencyProfile. e.g. KubeletConfig object used for updating Kubelet arguments
+	// +optional
+	RelatedObjects []ObjectReference `json:"relatedObjects,omitempty"`
```

Comment:
Why this is needed here?

Reply:
Taking MCO as an example, but this might be true with `Kube Controller Manager Operator` and `Kube API Server Operator` with their respective operations related to `WorkerLatencyProfile`, 

1. If the `WorkerLatencyProfile` is deleted, then MCO needs to also delete the associated `KubeletConfig` 
2. MCO controller can get evicted in the middle of `WorkerLatecyProfile` rollout, MCO needs to know which `KubeletConfig` was associated the given `WorkerLatencyProfile` in a persistent way. 

In other words, there has to be a correlation maintained between the `WorkerLatencyProfile` and `KubeletConfig` (in case of MCO) in a persistent way.

Reply:
Not quite sure what you mean with that association, every controller and operator (being also a controller) should read the current state of the system and update it according to what is represented in spec. For that you don't need that association, you should always work with the current state, no matter what happened before, no?

Reply:
I removed `RelatedObjects`, thanks.